nixos/modules/services/monitoring/prometheus/exporters/unbound.nix

```diff
@@ -0,0 +1,72 @@
+{ config, lib, pkgs, options }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.exporters.unbound;
+in
+{
+  port = 9167;
+  extraOpts = {
+    fetchType = mkOption {
+      type = types.enum [ "shm" "tcp" "uds" ];
+      default = "shm";
+      description = ''
+        Which methods the exporter uses to get the information from unbound.
+      '';
+    };
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = ''
+        Address under which the exporter exposes the metrics.
+      '';
+    };
+
+    telemetryPath = mkOption {
+      type = types.str;
+      default = "/metrics";
+      description = ''
+        Path under which to expose metrics.
+      '';
+    };
+
+    socketPath = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "/run/unbound/unbound.socket";
+      description = ''
+        Path to the unbound socket
+      '';
+    };
+
+    extraParams = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      example = [ "--log-level" ];
+      description = ''
+        Additional parameters passed to <command>unbound-telemetry</command>.
+      '';
+    };
+  };
+
+  serviceOpts = mkMerge ([{
+    serviceConfig = {
+      # User = "unbound";
```

Comment:
Is it necessary to run the service with this user in any case?

Reply:
I think it would be required in shm mode which is not fully implemented.

Reply:
Accessing the shared memory segment seems to work already without a normal user, with `DynamicUser` enabled. As mentioned, metric collection from shared memory is not yet implemented.